src/nodedef.h

```diff
@@ -493,9 +493,16 @@ struct ContentFeatures
 	bool isLiquid() const{
 		return (liquid_type != LIQUID_NONE);
 	}
-	bool sameLiquid(const ContentFeatures &f) const{
-		if(!isLiquid() || !f.isLiquid()) return false;
-		return (liquid_alternative_flowing_id == f.liquid_alternative_flowing_id);
+
+	bool isLiquidRender() const{
+		return (drawtype == NDT_LIQUID || drawtype == NDT_FLOWINGLIQUID);
+	}
+	bool sameLiquidRender(const ContentFeatures &f) const{
+		if (!isLiquidRender() || !f.isLiquidRender())
+			return false;
+		if (name == f.name)
+		       return true;
```

Comment:
Aren't these checks superfluous? Non-liquids have the lowing ID set to `CONTENT_IGNORE`, hence the last line should suffice and is faster on top of that.